dashboard/app/models/school_info.rb

```diff
@@ -159,12 +149,25 @@ def validate_non_us
   end
 
   def validate_us
+    if school || (school_type && school_type != SCHOOL_TYPE_NO_SCHOOL_SETTING)
```

Comment:
I'm going back and forth on whether to keep the right side of this OR statement. removing it broke a lot of existing tests due to their setup. honestly I think most of them could probably be deleted, as they were checking combinations where the associated nces school did not exist, but I think that's impossible with the new model and the data we capture now